bridge/ffi/src/lib.rs

```diff
@@ -9,12 +9,42 @@
 //! To use the FFI layer, you need to build the dynamic library (e.g., .so, .dylib, or .dll) by adding
 //! the appropriate configuration to `Cargo.toml`. Then, include the generated dynamic library and
 //! the C header file `dsnp_graph_sdk_ffi.h` in your C, C++, or other FFI-compatible projects.
+
 mod c_api;
 pub use c_api::*;
 mod bindings;
 pub use bindings::*;
 mod utils;
 pub use utils::*;
+mod errors;
+pub use errors::*;
 
 #[cfg(test)]
 mod tests;
+
+use std::ptr::NonNull;
+
+#[repr(C)]
+#[derive(Debug, Clone, PartialEq)]
+pub struct DsnpGraphFFIResult<T, E> {
+	pub result: Option<NonNull<T>>,
```

Comment:
What are the advantages of `NonNull` ?

Reply:
few that i learnt
* NonNull can be used to represent *mut/ *const types 
* Provides a safe abstraction for raw pointer (its essentially is kind of a smart pointer)
* NonNull types ensure the underlying pointer is never a null pointer and wrapping it around Option gives us ability to represent `NULL` type on c side while also being safe of rust side
* In general safer and sound in comparison to putting raw pointers there

Reply:
It seems like only one of `result`, `error` will ever be `Some(value)`, while the other would be `None`. If that's the case, would it make sense for this to be a union instead of a struct?